hilti/toolchain/src/compiler/visitors/resolver.cc

```diff
@@ -928,36 +928,57 @@ std::vector<Node> Visitor::matchOverloads(const std::vector<Operator>& candidate
         return r;
     };
 
-    for ( auto style : styles ) {
-        if ( disallow_type_changes )
-            style |= CoercionStyle::DisallowTypeChanges;
+    auto try_all_candidates = [&](std::vector<Node>* resolved, std::set<operator_::Kind>* kinds_resolved,
+                                  operator_::Priority priority) {
+        for ( auto style : styles ) {
+            if ( disallow_type_changes )
+                style |= CoercionStyle::DisallowTypeChanges;
 
-        HILTI_DEBUG(logging::debug::Operator, util::fmt("style: %s", to_string(style)));
-        logging::DebugPushIndent _(logging::debug::Operator);
+            HILTI_DEBUG(logging::debug::Operator, util::fmt("style: %s", to_string(style)));
+            logging::DebugPushIndent _(logging::debug::Operator);
 
-        std::vector<Node> resolved;
+            for ( const auto& c : candidates ) {
+                if ( priority != c.priority() )
+                    // Not looking at operators of this priority right now.
+                    continue;
 
-        for ( const auto& c : candidates ) {
-            HILTI_DEBUG(logging::debug::Operator, util::fmt("candidate: %s", c.typename_()));
-            logging::DebugPushIndent _(logging::debug::Operator);
+                if ( priority == operator_::Priority::Low && kinds_resolved->find(c.kind()) != kinds_resolved->end() )
+                    // Already have a higher priority match for this operator kind.
+                    continue;
```

Comment:
We already enforce some ordering outside, suggest to drop the check for `Low` here,
```suggestion
                if ( kinds_resolved->find(c.kind()) != kinds_resolved->end() )
                    // Already have a previous match for this operator kind.
                    continue;
```

Reply:
You are right, we mutate `kinds` below.